--- openssl-1.0.1f.orig/apps/openssl.c	2014-01-06 13:47:42.000000000 +0000
+++ openssl-1.0.1f/apps/openssl.c	2014-02-21 03:45:18.889381156 +0000
@@ -345,8 +345,10 @@ int main(int Argc, char *ARGV[])
 		if (ERR_GET_REASON(ERR_peek_last_error())
 		    == CONF_R_NO_SUCH_FILE)
 			{
+#if 0 /* ANDROID */
 			BIO_printf(bio_err,
 				   "WARNING: can't open config file: %s\n",p);
+#endif
 			ERR_clear_error();
 			NCONF_free(config);
 			config = NULL;
--- openssl-1.0.1f.orig/apps/progs.h	2014-01-06 13:47:42.000000000 +0000
+++ openssl-1.0.1f/apps/progs.h	2014-02-21 03:45:36.389880972 +0000
@@ -147,7 +147,9 @@ FUNCTION functions[] = {
 	{FUNC_TYPE_GENERAL,"ocsp",ocsp_main},
 #endif
 	{FUNC_TYPE_GENERAL,"prime",prime_main},
+#if 0 /* ANDROID */
 	{FUNC_TYPE_GENERAL,"ts",ts_main},
+#endif
 #ifndef OPENSSL_NO_SRP
 	{FUNC_TYPE_GENERAL,"srp",srp_main},
 #endif
--- openssl-1.0.1f.orig/apps/speed.c	2014-01-06 13:47:42.000000000 +0000
+++ openssl-1.0.1f/apps/speed.c	2014-02-21 03:45:18.893380356 +0000
@@ -1746,6 +1746,7 @@ int MAIN(int argc, char **argv)
 			}
 		}
 
+#if 0 /* ANDROID */
 	if (doit[D_IGE_128_AES])
 		{
 		for (j=0; j<SIZE_NUM; j++)
@@ -1806,6 +1807,7 @@ int MAIN(int argc, char **argv)
 		}
 
 #endif
+#endif
 #ifndef OPENSSL_NO_CAMELLIA
 	if (doit[D_CBC_128_CML])
 		{
--- openssl-1.0.1f.orig/crypto/ui/ui_openssl.c	2014-01-06 13:47:42.000000000 +0000
+++ openssl-1.0.1f/crypto/ui/ui_openssl.c	2014-02-21 03:45:18.897379557 +0000
@@ -190,7 +190,7 @@
 # undef  SGTTY
 #endif
 
-#if defined(linux) && !defined(TERMIO)
+#if defined(linux) && !defined(TERMIO) && !defined(__ANDROID__)
 # undef  TERMIOS
 # define TERMIO
 # undef  SGTTY
